protocol/protobuf/url_data.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+option go_package = "./;protobuf";
+package protobuf;
+
+message Community {
+ string display_name = 1;
+ string description = 2;
+ uint32 members_count = 3;
+ string color = 4;
+ repeated uint32 tag_indices = 5;
+}
+
+message Channel {
+ string display_name = 1;
+ string description = 2;
+ string emoji = 3;
+ string color = 4;
+ Community community = 5;
+ string uuid = 6;
+}
+
+message User {
+ string display_name = 1;
+ string description = 2;
+ string color = 3;
+}
+
+message URLData {
+ // Community, Channel, or User
+ bytes content = 1;
+}
+
+// Field on CommunityDescription, CommunityChat and ContactCodeAdvertisement
+message URLParams {
+ string encoded_url_data = 1;
+ // Signature of encoded URL data
+ string encoded_signature = 2;
+}
```

Comment:
It's generally accepted that we should end files with *proper* lines. If a reference helps, according to the POSIX standard, proper lines should have a terminating newline character. If you allow me, I recommend configuring your text editor of choice to do this for you. All of them have this feature.

https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206